### PR TITLE
Install packages from cadl package and list installed packages

### PR DESCRIPTION
**Pull request by praveenkuttappan**
CADL package should be installed from package.json and also added a debug line to show installed packages.

### CODE REVIEW

**Comment by praveenkuttappan** [comment]
This PR will resolve issue of API review not getting created for CADL even though token file is generated.

**Review by tjprescott — CHANGES_REQUESTED** [review]
You will still need to install cadl-apiview (soon to be typespec-apiview) if the package.json does not list it in order for APIView generation to work.

**Comment by praveenkuttappan** [comment]
> You will still need to install cadl-apiview (soon to be typespec-apiview) if the package.json does not list it in order for APIView generation to work.

cadl-apiview will get installed when installing packages using global package.json and that will cover the scenario of not having this dependency in project specific package.json.

**Review by tjprescott — APPROVED** [review]

File: eng/scripts/Create-Apiview-Token-Cadl.ps1 [hunk 1]
```diff
@@ -19,18 +20,29 @@ function Sparse-Checkout($branchName, $packagePath)
...
     Write-Host "Generating API review token file from path '$($packagePath)'"
     Push-Location $packagePath
     try
     {
+
+        # Install required packages using project specific package.json          
+        npm install
+        npm list
+        # Install additional/missing packages using gloal package.json
+        Rename-Item package.json package-orig.json
```
Comment (weshaggard) [inline eng/scripts/Create-Apiview-Token-Cadl.ps1:40]: Is this just for testing or temporary until we get the correct global settings?

File: eng/scripts/Create-Apiview-Token-Cadl.ps1 [hunk 2]
```diff
@@ -11,6 +11,7 @@ param (
 
 Set-StrictMode -Version 3
 $sparseCheckoutFile = ".git/info/sparse-checkout"
+$globalPackageJsonPath = "specification/package.json"
```
Comment (weshaggard) [inline eng/scripts/Create-Apiview-Token-Cadl.ps1:14]: Note this is going to change to the root of the repo soon in https://github.com/Azure/azure-rest-api-specs/pull/22735.

File: eng/scripts/Create-Apiview-Token-Cadl.ps1 (hunk 1 above)
Comment (weshaggard) [reply eng/scripts/Create-Apiview-Token-Cadl.ps1:40]: Or are you trying to enable both sets of package installs?

Comment (praveenkuttappan) [reply eng/scripts/Create-Apiview-Token-Cadl.ps1]: Done.

File: eng/scripts/Create-Apiview-Token-Cadl.ps1 [hunk 3]
```diff
@@ -19,24 +20,19 @@ function Sparse-Checkout($branchName, $packagePath)
...
-        npm install
-        cadl compile . --emit=@azure-tools/cadl-apiview
-    }
-    finally
-    {
-        Pop-Location
-    }
+    npm install
+    npm list
+    cadl compile . --emit=@azure-tools/cadl-apiview
```
Comment (weshaggard) [inline eng/scripts/Create-Apiview-Token-Cadl.ps1:34]: Doesn't this also require that you add this package into that global package.json to get it to work?

Comment (weshaggard) [reply eng/scripts/Create-Apiview-Token-Cadl.ps1:34]: Nevermind it is already added https://github.com/Azure/azure-rest-api-specs/blob/main/package.json#L17

**Pull request merged**